test/Streamly/Test/Data/Unbox.hs

```diff
@@ -160,21 +209,32 @@ testGenericConsistency ::
 testGenericConsistency val = do
 
     -- Test the generic sizeOf
-    sizeOf (Proxy :: Proxy a) `shouldBe` genericSizeOf (Proxy :: Proxy a)
+#ifdef USE_SERIALIZE
+    (case size :: Size a of
+         ConstSize x -> x
+         VarSize f -> f val)
+#else
+    sizeOf (Proxy :: Proxy a)
+#endif
+               `shouldBe` genericSizeOf (Proxy :: Proxy a)
 
     -- Test the serialization and deserialization
     arr <- newBytes (sizeOf (Proxy :: Proxy a))
 
-    pokeByteIndex 0 arr val
+    POKE(0, arr, val)
     genericPeekByteIndex arr 0 `shouldReturn` val
 
     genericPokeByteIndex arr 0 val
-    peekByteIndex 0 arr `shouldReturn` val
+    PEEK(0, arr) `shouldReturn` val
+
 
+#ifndef USE_SERIALIZE
 -- Size is also implicitly tested while serializing and deserializing.
 checkSizeOf :: forall a. Unbox a => Proxy a -> Int -> IO ()
 checkSizeOf _ size = sizeOf (Proxy :: Proxy a) `shouldBe` size
```

Comment:
We can do this for Serialize case as well, at least for the const size case.

Reply:
I abuse `VarSize`. I'm thinking `serialize` should only have `Varsize`. Or it shouldn't have `size` at all. Somehow return a stream of arrays.